docs/stake-optimizer.md

```diff
@@ -0,0 +1,260 @@
+When you delegate tokens to a validator for a determined period, you can use the [`auto-compounder`](https://github.com/fetchai/cosmpy/blob/develop/examples/aerial_compounder.py) to get increasing rewards. You can maximize your rewards for a given staking period by selecting an optimal compounding period. To do this, you will need to follow these steps:
+
+* **Set and Query Variables**: When calculating staking rewards, you need to set and query variables such as staking parameters, transaction fees, and network parameters
+* **Calculate Reward Rate**: After you select and query all the variables needed, you will calculate the reward rate
+* **Calculate Optimal Compounding Period**: You will calculate the optimal compounding period that will maximize your rewards
+
+First, you need to define a network to work with, in this case, we will interact with dorado testnet.
+
+```python
+from cosmpy.aerial.client import LedgerClient
+from cosmpy.aerial.config import NetworkConfig
+
+ledger = LedgerClient(NetworkConfig.fetchai_dorado_testnet())
```

Comment:
I think this generic function is better to use here: https://github.com/fetchai/cosmpy/blob/develop/examples/aerial_oracle.py#L56.